mail_restrict_follower_selection/models/mail_thread.py

```diff
@@ -23,18 +23,18 @@ def _message_add_suggested_recipient(
             "mail.wizard.invite"
         ]._mail_restrict_follower_selection_get_domain()
         eval_domain = safe_eval(
-            domain, locals_dict={"ref": lambda str_id: _id_get(self.env, str_id)}
+            str(domain), locals_dict={"ref": lambda str_id: _id_get(self.env, str_id)}
         )
-        for key in result:
-            items_to_remove = []
-            for item in result[key]:
-                partner_id = item[0]
-                if partner_id:
-                    partner = self.env["res.partner"].search(
-                        [("id", "=", partner_id)] + eval_domain
-                    )
-                    if not partner:
-                        items_to_remove.append(item)
-            for item in items_to_remove:
-                result[key].remove(item)
+        items_to_remove = []
+        for item in result:
+            partner_id = item.get("partner_id", False)
+            if partner_id:
+                partner_count = self.env["res.partner"].search_count(
+                    [("id", "=", partner_id)] + eval_domain
+                )
+                if not partner_count:
+                    items_to_remove.append(item)
+        for item in items_to_remove:
+            result.remove(item)
+
```

Comment:
Please add a note regarding this structural change. 
Hint: this change originated from [Here](https://github.com/odoo/odoo/commit/7a32c151e3e8e62c0a6b74b05b49494f1d0f24b8#diff-a90153cd0202b1feb0d5ef9cf041bf662256cb7ce1434b610709a7557db3c9ceR1803-R1833)